h2o-py/h2o/explanation/_explain.py

```diff
@@ -951,33 +951,100 @@ def mapping(x):
         plt.xticks(mapping(range(nf.nlevels(column))), nf.levels(column))
     plt.ylim(ylims)
 
-def _handle_ice(model, frame, colormap, plt, target, is_factor, column, show_logodds, centered, factor_map, show_pdp, **kwargs):
+
+def _append_graphing_data(graphing_data, data_to_append, original_observation_value, frame_id, centered, show_logoods,
+                          row_id, **kwargs):
+    grouping_variable_value = kwargs.get("grouping_variable_value", None)
+    response_type = data_to_append.col_types[data_to_append.col_header.index("mean_response")]
+    grouping_variable_type = "string" if type(grouping_variable_value) is str else "double" # todo test this
+    bin_type = data_to_append.col_types[0]
+
+    col_header = ["sample_id", "row_id", "column", "mean_response", "simulated_x_value", "is_original_observation"]
+    col_types = ["string", "int", "string", response_type, bin_type, "bool"]
+    table_header = "ICE plot graphing output" + kwargs.get("group_label", "")
+
+    if grouping_variable_value is not None:
+        col_header.append("grouping_variable_value")
+        col_types.append(grouping_variable_type)
+    centering_value = None
+    if centered:
+        col_header.append("centered_response")
+        col_types.append(response_type)
+        centering_value = data_to_append['mean_response'][0]
+    if show_logoods:
+        col_header.append("log(odds)")
+        col_types.append(response_type)
+
+    if graphing_data is None:
+        return h2o.two_dim_table.H2OTwoDimTable(col_header=col_header, col_types=col_types, table_header=table_header,
+                                                cell_values=_extract_graphing_data_values(data_to_append, frame_id,
+                                                                                          grouping_variable_value,
+                                                                                          original_observation_value,
+                                                                                          centering_value, show_logoods,
+                                                                                          row_id))
+    if data_to_append is None:
+        return graphing_data
```

Comment:
It's not really clear how these 2 situations can happen. As a spectator without the context, I'd say the `graphing_data` and `data_to_append` at least deserve a docstring, explaining what they do.
Also the function is called `_append_graphing_data`, but it's not clear what is appended where.

Reply:
did some documentation in [f2bd46e](https://github.com/h2oai/h2o-3/pull/6119/commits/f2bd46e90cf03fc39da9cffae0cbe66a7064279c)